httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/HappyEyeballsV2AsyncClientConnectionOperatorBuilder.java

```diff
@@ -0,0 +1,272 @@
+     * @return this builder
+     */
+    public HappyEyeballsV2AsyncClientConnectionOperatorBuilder withFirstAddressFamilyCount(
+            final int firstAddressFamilyCount) {
+        this.firstAddressFamilyCount = firstAddressFamilyCount;
+        return this;
+    }
+
+    /**
+     * Sets the preferred address family to use for connections.
+     *
+     * @param addressFamily the preferred address family
+     * @return this builder
+     */
+    public HappyEyeballsV2AsyncClientConnectionOperatorBuilder withAddressFamily(final HappyEyeballsV2AsyncClientConnectionOperator.AddressFamily addressFamily) {
+        this.addressFamily = addressFamily;
+        return this;
+    }
+
+    /**
+     * Builds a {@link HappyEyeballsV2AsyncClientConnectionOperator} with the specified parameters.
+     *
+     * @return the {@link HappyEyeballsV2AsyncClientConnectionOperator} instance built with the specified parameters.
+     * @throws IllegalArgumentException if the connection operator is null.
+     */
+    public HappyEyeballsV2AsyncClientConnectionOperator build() {
+        final TlsStrategy tlsStrategyCopy;
+        if (tlsStrategy != null) {
+            tlsStrategyCopy = tlsStrategy;
+        } else {
+            if (ReflectionUtils.determineJRELevel() <= 8 && ConscryptClientTlsStrategy.isSupported()) {
+                if (systemProperties) {
+                    tlsStrategyCopy = ConscryptClientTlsStrategy.getSystemDefault();
+                } else {
+                    tlsStrategyCopy = ConscryptClientTlsStrategy.getDefault();
+                }
+            } else {
+                if (systemProperties) {
+                    tlsStrategyCopy = DefaultClientTlsStrategy.getSystemDefault();
+                } else {
+                    tlsStrategyCopy = DefaultClientTlsStrategy.getDefault();
+                }
+            }
+        }
+
+
+        connectionOperator = Args.notNull(connectionOperator, "Connection operator");
+        final DnsResolver dnsResolverToUse = dnsResolver != null ? dnsResolver : SystemDefaultDnsResolver.INSTANCE;
+        final Timeout timeoutToUse = timeout != null ? timeout : Timeout.ofMilliseconds(250);
+        final Timeout resolutionDelayToUse = resolutionDelay != null ? resolutionDelay : Timeout.ofMilliseconds(50);
+        final Timeout minimumConnectionAttemptDelayToUse = minimumConnectionAttemptDelay != null ? minimumConnectionAttemptDelay : Timeout.ofMilliseconds(100);
+        final Timeout maximumConnectionAttemptDelayToUse = maximumConnectionAttemptDelay != null ? maximumConnectionAttemptDelay : Timeout.ofSeconds(2);
+        final Timeout connectionAttemptDelayToUse = connectionAttemptDelay != null ? connectionAttemptDelay : Timeout.ofMilliseconds(250);
+        final int firstAddressFamilyCountToUse = Args.positive(firstAddressFamilyCount, "First Address Family");
+        final HappyEyeballsV2AsyncClientConnectionOperator.AddressFamily addressFamilyToUse = addressFamily;
```

Comment:
These defaults are all duplicated in the `HappyEyeballsV2AsyncClientConnectionOperator` constructor.